### ISSUE TITLE
Bug(`prefer-string-schema-with-trim`): Messing up mini

### ISSUE THREAD

**Issue by demosjarco** [marcalexiei/eslint-plugin-zod#215]
### 🧩 Description

The rule doesn't detect `.trim()` inside a `.check()` as is the proper format of mini.
Also auto fix adds it as direct dot notation rather than inside check() and it breaks the mini schema.
<img width="362" height="56" alt="Image" src="https://github.com/user-attachments/assets/94fcab95-cc26-4b49-bee0-c19077537467" />
Disabling the rule is sub-optimal since this is a mixed zod 4/mini project (I try to use mini wherever possible, but other libraries require/or use 4).

### 🪜 How to reproduce

1. Make a zod mini schema without `trim()`, or `trim()` inside `check()`.
2. Run eslint with the recommended config

### ✅ Expected behavior

Add in `trim()` correctly as per mini spec. Or if too complicated/not supported, to at least detect if `mini` is used in that schema and skip it rather than breaking it.

### 💻 System info

```shell
Binaries:
    Node: 24.14.0
    npm: 11.9.0
```
